kroxylicious-filters/kroxylicious-record-validation/src/test/java/io/kroxylicious/proxy/filter/schema/JsonSchemaBytebufValidatorTest.java

```diff
@@ -98,63 +98,96 @@ public static void shutdownMockRegistry() {
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(true, Result::valid);
     }
 
     @Test
-    void testInvalidValueInValidated() {
-        Record record = createRecord("a", "123");
-        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(Map.of("apicurio.registry.url", registryServer.baseUrl()), 1L);
-        Result result = validate(record, validator);
-        assertFalse(result.valid());
-        verifyNoMoreInteractions(mockValidator);
+    void valueWithCorrectSchemaIdInBodyPassesValidation() {
+        var value = asSchemaIdPrefixBuf(GLOBAL_ID, VALID_JSON);
+        Record record = record(RECORD_KEY, value);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(true, Result::valid);
     }
 
-    private static Result validate(Record record, BytebufValidator validator) {
-        try {
-            return validator.validate(record.value(), record, false).toCompletableFuture().get(5, TimeUnit.SECONDS);
-        }
-        catch (ExecutionException | InterruptedException | TimeoutException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    void valueWithWrongSchemaIdInHeaderRejected() {
+        Header[] headers = new Header[]{ new RecordHeader(SerdeHeaders.HEADER_VALUE_GLOBAL_ID, toByteArray(GLOBAL_ID + 1)) };
+        Record record = record(RECORD_KEY, VALID_JSON, headers);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(false, Result::valid)
+                .returns("Unexpected schema id in record (2), expecting 1", Result::errorMessage);
     }
 
-    private Record createRecord(String key, String value) {
-        ByteBuffer keyBuf = toBufNullable(key);
-        ByteBuffer valueBuf = toBufNullable(value);
-
-        try (ByteBufferOutputStream bufferOutputStream = new ByteBufferOutputStream(1000); DataOutputStream dataOutputStream = new DataOutputStream(bufferOutputStream)) {
-            DefaultRecord.writeTo(dataOutputStream, 0, 0, keyBuf, valueBuf, Record.EMPTY_HEADERS);
-            dataOutputStream.flush();
-            bufferOutputStream.flush();
-            ByteBuffer buffer = bufferOutputStream.buffer();
-            buffer.flip();
-            return DefaultRecord.readFrom(buffer, 0, 0, 0, 0L);
-        }
-        catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    void valueWithWrongSchemaIdInBodyRejected() {
```

Comment:
```suggestion
    void valueWithUnexpectedSchemaIdInBodyRejected() {
```
As it could be `Wrong` for a variety of reasons.